config.go

```diff
@@ -1917,6 +1925,52 @@ func parseRPCParams(cConfig *lncfg.Chain, nodeConfig interface{},
 	return nil
 }
 
+// extractEnvVariable extracts the value of an environment variable from a string.
+// It supports the following formats:
+// 1) $ENVIRONMENT_VARIABLE
+// 2) ${ENVIRONMENT_VARIABLE}
+// 3) ${ENVIRONMENT_VARIABLE:-DEFAULT_VALUE}
+//
+// Standard environment variable naming conventions:
+// - ENVIRONMENT_VARIABLE contains letters, digits, and underscores, and does not start with a digit.
+// - DEFAULT_VALUE follows the rule that it can contain any characters except whitespace.
+//
+// Parameters:
+// - value: The input string containing references to environment variables.
+//
+// Returns:
+//   - string: The value of the specified environment variable, or the default value
+//     if provided, or the original input string if no matching variable is found or set.
+func extractEnvVariable(value string) string {
+	re := regexp.MustCompile(`^(?:\${([a-zA-Z_][a-zA-Z0-9_]*)?(:-(\S+))?})|(?:\$([a-zA-Z_][a-zA-Z0-9_]*))$`)
```

Comment:
When I plug this into a [diagram tool](https://regexper.com/#%5E%28%3F%3A%5C%24%7B%28%5Ba-zA-Z_%5D%5Ba-zA-Z0-9_%5D*%29%3F%28%3A-%28%5CS%2B%29%29%3F%7D%29%7C%28%3F%3A%5C%24%28%5Ba-zA-Z_%5D%5Ba-zA-Z0-9_%5D*%29%29%24), I notice that the start of line and end of line characters are broken up across the branches. I don't believe you intended this. I would suggest making it so that both branches are anchored at the beginning and end of line.

Reply:
Ah yes, this makes sense and thanks a lot for mentioning this diagram tool the regular expressions make more sense now :)

Here is after updating the regular expression so that both branches are anchored at the beginning and end of line.
<img width="1320" alt="Screenshot 2024-01-05 at 7 03 46 PM" src="https://github.com/lightningnetwork/lnd/assets/69568555/ae263b74-9dcc-4142-9579-db3c36ff6fce">

Reply:
Yeah I wasn't going to eyeball that regex and be like "LGTM" 😅